matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/auth/version/HomeServerVersion.kt

```diff
@@ -38,14 +38,14 @@ internal data class HomeServerVersion(
     }
 
     companion object {
-        internal val pattern = Regex("""[r|v](\d+)\.(\d+)\.(\d+)""")
+        internal val pattern = Regex("""[r|v](\d+)\.(\d+)(?:\.(\d+))?""")
```

Comment:
this is the fix, to make the last capture group optional

Reply:
It should work, but strange that it's not just `[r|v](\d+)\.(\d+)(\.(\d+))?`. Regex syntax I guess.